src/main/webapp/app/shared/user-settings/user-settings-prototype/user-settings-prototype.component.ts

```diff
@@ -0,0 +1,106 @@
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { JhiAlertService } from 'ng-jhipster';
+import { User } from 'app/core/user/user.model';
+import { UserSettingsCategory } from 'app/shared/constants/user-settings.constants';
+import { OptionCore, UserSettings } from 'app/shared/user-settings/user-settings.model';
+import { UserSettingsService } from 'app/shared/user-settings/user-settings.service';
+
+/**
+ * Is used as the abstract user-settings "parent" component with all the necessary basic logic for other "child" components to implement/inherit from.
+ */
+@Component({
+    templateUrl: 'user-settings-prototype.component.html',
```

Comment:
(i) How does this relate to the file: _"user-settings-prototype.component.html"_ . I see that file is commented.
(i) Do we even need the `@Component` directive in this case, if the template is not being used.

Reply:
![image](https://user-images.githubusercontent.com/65814168/132983530-58c05b49-807c-47d9-b45e-ade266911dd0.png)

As far as I know I sadly have to use the @Component decorator and a valid templateUrl, otherwise error occur during compilation, because I am using OnInit and I can only do so if the class has such a decorator.

The reason why the html is commented out is due to the reason that it is not possible to reuse this template in a polymorphic/inheritance kind of way due to the different implementation of the concrete OptionCore and its bidirectional connectivity to the template (e.g. you need to access the option that is only possible due to the ngFor loop but this creates huge problems for reusability. Therefore I commented this template out to provide an examplary template that can be easily copied for new settings.

![image](https://user-images.githubusercontent.com/65814168/132983549-a2dc811d-c439-49c2-a83a-bb11f4d4c6e0.png)